plugins/flatpak/gs-flatpak.c

```diff
@@ -4068,8 +4068,17 @@ gs_flatpak_update_app (GsFlatpak *self,
 		}
 
 		list = gs_app_list_new ();
-		for (guint i = 0; i < proxied_apps->len; ++i)
-			gs_app_list_add (list, g_ptr_array_index (proxied_apps, i));
+		for (guint i = 0; i < proxied_apps->len; ++i) {
+			g_autoptr(GsAppList) app_related_list = NULL;
+			GsApp *proxied_app = g_ptr_array_index (proxied_apps, i);
+
+			/* Add not only the app but also any eventual related apps to be
+			 * updated; otherwise, the related apps of proxied apps may never
+			 * get updated. */
+			app_related_list = gs_flatpak_get_list_for_update (self, proxied_app,
+									   cancellable, error);
```

Comment:
You are not handling the case where this call fails and `error` is set.

Reply:
Rushy rushy! I will address it.